activerecord/test/cases/associations/eager_test.rb

```diff
@@ -1511,6 +1511,49 @@ def test_preloading_has_many_through_with_custom_scope
     Author.preload(:readonly_comments).first!
   end
 
+  # Associations::Preloader#preloaders_on works with hash-like objects
+  test "preloading works when object that responds to :to_hash is passed" do
+    post_id = Comment.where(author_id: nil).where.not(post_id: nil).first.post_id
+    CustomHash = Class.new(Hash)
+
+    assert_nothing_raised do
+      Post.preload(CustomHash.new(comments: [{ author: :essays }])).find(post_id)
+    end
+  end
+
+  # Associations::Preloader#preloaders_on works with symbol-like objects
+  test "preloading works when object that responds to :to_sym is passed" do
+    post_id = Comment.where(author_id: nil).where.not(post_id: nil).first.post_id
+    custom_symbol = (1..10)
```

Comment:
If this specific range is not important to the test, you can use `Object.new`. You could also just use a string or some other object that already responds to `to_sym`.